CMakeLists.txt

```diff
@@ -261,3 +263,95 @@ if(CELERITAS_USE_Doxygen)
+# CONFIG FILE INSTALLATION
+#----------------------------------------------------------------------------#
+
+# Where to configure cmake files
+set(CELER_CMAKE_CONFIG_DIRECTORY "${PROJECT_BINARY_DIR}/share")
+# Where to install configured cmake files
+set(CELER_INSTALL_CMAKECONFIGDIR
+  "${CMAKE_INSTALL_DATAROOTDIR}/cmake/Celeritas")
+
+# Build list of CMake files to install
+set(_cmake_files
+  "${PROJECT_SOURCE_DIR}/cmake/CeleritasGen.cmake"
+  "${PROJECT_SOURCE_DIR}/cmake/CeleritasGen/gen-action.py"
+  "${PROJECT_SOURCE_DIR}/cmake/CeleritasLibrary.cmake"
+)
+foreach(_dep Geant4 HepMC3 ROOT VecGeom)
+  if(CELERITAS_USE_${_dep})
+    list(APPEND _cmake_files "${PROJECT_SOURCE_DIR}/cmake/Find${_dep}.cmake")
+  endif()
+endforeach()
+if(CELERITAS_BUILD_TESTS)
+  list(APPEND _cmake_files
+    "${PROJECT_SOURCE_DIR}/cmake/CeleritasAddTest.cmake"
+  )
+endif()
+
+install(FILES ${_cmake_files}
+  DESTINATION "${CELER_INSTALL_CMAKECONFIGDIR}"
+  COMPONENT devel
+)
+install(DIRECTORY "${PROJECT_SOURCE_DIR}/cmake/backport"
+  DESTINATION "${CELER_INSTALL_CMAKECONFIGDIR}"
+  COMPONENT devel
+)
+
+# Export all cache variables that start with CELERITAS_
+set(CELER_EXPORT_VARIABLES)
+get_directory_property(_cachevar_keys CACHE_VARIABLES)
+foreach(_key IN LISTS _cachevar_keys)
+  if(_key MATCHES "^CELERITAS_")
+    list(APPEND CELER_EXPORT_VARIABLES "set(${_key} \"${${_key}}\")")
+  endif()
+endforeach()
+
+# Add other cache variables, prefixed with CELER_
+foreach(_key BUILD_SHARED_LIBS)
+  list(APPEND CELER_EXPORT_VARIABLES "set(CELERITAS_${_key} \"${${_key}}\")")
+endforeach()
+
+# Add hints for direct dependencies and indirect geant dependencies
+foreach(_key
+  MPIEXEC_EXECUTABLE nlohmann_json_DIR VecCore_DIR VecGeom_DIR Python_DIR Geant4_DIR
+  CLHEP_DIR ZLIB_DIR EXPAT_DIR XercesC_DIR PTL_DIR
+  EXPAT_INCLUDE_DIR EXPAT_LIBRARY
+  XercesC_LIBRARY XercesC_INCLUDE_DIR
+)
+  set(_val "${${_key}}")
+  if(_val)
+    list(APPEND CELER_EXPORT_VARIABLES "set_and_check(${_key} \"${_val}\")")
+  endif()
+endforeach()
+list(JOIN CELER_EXPORT_VARIABLES "\n" CELER_EXPORT_VARIABLES)
+
+# Generate the file needed by downstream "find_package(CELER)"
+configure_package_config_file(
+  "${PROJECT_SOURCE_DIR}/cmake/CeleritasConfig.cmake.in"
+  "${CELER_CMAKE_CONFIG_DIRECTORY}/CeleritasConfig.cmake"
+  INSTALL_DESTINATION "${CELER_INSTALL_CMAKECONFIGDIR}"
+)
+
+# Export version info
+write_basic_package_version_file(
+  "${CELER_CMAKE_CONFIG_DIRECTORY}/CeleritasConfigVersion.cmake"
+  COMPATIBILITY ExactVersion
+)
+
+# Install the config and version files
+install(FILES
+  "${CELER_CMAKE_CONFIG_DIRECTORY}/CeleritasConfig.cmake"
+  "${CELER_CMAKE_CONFIG_DIRECTORY}/CeleritasConfigVersion.cmake"
+  DESTINATION ${CELER_INSTALL_CMAKECONFIGDIR}
+)
+
+# Install 'CeleritasTargets.cmake', included by CeleritasConfig.cmake, which
+# references the targets we install.
+install(EXPORT celeritas-targets
+  FILE CeleritasTargets.cmake
+  NAMESPACE ${CELER_NAMESPACE}
```

Comment:
Is `CELER_NAMESPACE` going to be set by the project? This would save the need to create the `ALIAS` target at the end of `CeleritasConfig.cmake`.

Reply:
Oops, I left that in while I was playing around with it. Unfortunately since we name our libraries `celeritas` and `celeritas_io`, the only way to make them work "naturally" with namespaces is to refer to them as `Celeritas::celeritas` and `Celeritas::celeritas_io`. Currently we refer to them as `Celeritas::Core` and `Celeritas::IO` to avoid the extra duplication.

If we end up building separate libraries for corecel/orange/celeritas, I think we should reinstate the namespaces—but even then, if we want to provide alias targets for IO/Core, we apparently still have to define them manually in the Config file: Targets.cmake won't pick them up.

Reply:
I'd actually still put the namespace in, even if creating the "short" `ALIASES` as official downstream targets. A namespaced target is never at risk of being converted to a raw `-l<name>` link which provides a _little_ bit of safety against misconfiguration (I'm not sure that having an `ALIAS` that's namespaced prevents this..

Reply:
That's true, but my thought is that *only* the aliased libraries (and yes it should prevent that) should be used by downstream code: the targets `celeritas` and `celeritas_io` should be considered private. I guess it can't hurt though! (I only avoided it because I think `Celeritas::celeritas` is ugly 😂 )

Reply:
Oh yeah, I also avoided the namespace because of the wackiness with the extra CUDA libraries needed for linking against vecgeom. I'll see if I can get it working...